code/modules/mob/living/carbon/carbon.dm

```diff
@@ -251,17 +251,22 @@
 					"<span class='notice'>You pat [src] on the head.</span>",\
 					)
 				else
-
-					M.visible_message(\
-					"<span class='notice'>[M] gives [src] a [pick("hug","warm embrace")].</span>",\
-					"<span class='notice'>You hug [src].</span>",\
-					)
-					if(ishuman(src))
-						var/mob/living/carbon/human/H = src
-						if(H.wear_suit)
-							H.wear_suit.add_fingerprint(M)
-						else if(H.w_uniform)
-							H.w_uniform.add_fingerprint(M)
+					if(M.zone_selected == ("r_arm" || "r_hand" || "l_arm" || "l_hand"))
+						M.visible_message(\
+						"<span class='notice'>[M] shake [src] hand.</span>",\
+						"<span class='notice'>You shake hands with [src].</span>",\
+						)
+					else
+						M.visible_message(\
+						"<span class='notice'>[M] gives [src] a [pick("hug","warm embrace")].</span>",\
+						"<span class='notice'>[M] You hug [src].</span>",\
+						)
```

Comment:
The formatting of the arguments is very odd here, but it's also like this in the rest of the proc too and this is your first PR here, so I think it's fine to stay this way for now.